internal/saucecloud/cloud.go

```diff
@@ -146,30 +143,15 @@ func (r *CloudRunner) collectResults(artifactCfg config.ArtifactDownload, result
 			}
 
 			var artifacts []report.Artifact
-			var parentJUnits []report.Artifact
-
-			if junitRequired {
-				jb, err := r.JobService.GetJobAssetFileContent(
-					context.Background(),
-					res.job.ID,
-					junit.JunitFileName,
-					res.job.IsRDC)
-				artifacts = append(artifacts, report.Artifact{
-					AssetType: report.JUnitArtifact,
-					Body:      jb,
-					Error:     err,
-				})
-				for _, id := range res.previous {
-					jb, err := r.JobService.GetJobAssetFileContent(
-						context.Background(),
-						id,
-						junit.JunitFileName,
-						res.job.IsRDC,
-					)
-					parentJUnits = append(parentJUnits, report.Artifact{
-						AssetType: report.JUnitArtifact,
-						Body:      jb,
-						Error:     err,
+			junitArtifact, parentJUnits := r.GetJUnitArtifacts(res)
+			artifacts = append(artifacts, junitArtifact)
+
+			files := r.downloadArtifacts(res.name, res.job, artifactCfg.When)
+			if report.IsArtifactRequired(r.Reporters, report.JSONArtifact) {
+				for _, f := range files {
+					artifacts = append(artifacts, report.Artifact{
+						FilePath:  f,
+						AssetType: report.JSONArtifact,
```

Comment:
So this is the part that seems off in the refactor. You're adding the list of artifacts that the user specified to download and assigning all of them the asset type of a json artifact. But how do you know that they are?

I think you're trying to fix another problem somewhere else. But exchanging one hack for another is not really the best way to do that 😅

Reply:
yep. This is introduced to solve [filtering artifacts](https://github.com/saucelabs/saucectl/blob/e36105e84dd037deba210ef88992c0fa59947489/internal/junit/reporter.go#L113)

Why?
The enum `Iota` is starting from (0)[https://github.com/saucelabs/saucectl/blob/main/internal/report/report.go#L30].
If default value of (ArtifactType)[https://github.com/saucelabs/saucectl/blob/main/internal/report/report.go#L26C13-L26C13] is `0`,  when filtering JUnit artifacts, those `report.Artifact` with default value would be treated as `JUnitArtifact`. That's not what we expected.

I can also define the enums like
```
const (
	// JUnitArtifact represents the junit artifact type (https://llg.cubic.org/docs/junit/).
	JUnitArtifact ArtifactType = iota + 1
	// JSONArtifact represents the json artifact type
	JSONArtifact
)

```

Or
```
const (
	DefaultArtifact ArtifactType = iota
	// JUnitArtifact represents the junit artifact type (https://llg.cubic.org/docs/junit/).
	JUnitArtifact
	// JSONArtifact represents the json artifact type
	JSONArtifact
)
```
The first approach is implicit version of the second one.

While considering the usage of `JSONArtifact`, is it really necessary? What kind of artifacts can be treated as `JSONArtifact`? For now, I can see it's only for filtering a JUnit artifact. Then I think a `DefaultArtifact` and a `JUnitArtifact` should be enough.